plugin/buycourses/src/cecabank_response.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/* For license terms, see /license.txt */
+
+/**
+ * Success page for the purchase of a course in the Buy Courses plugin.
+ *
+ * @package chamilo.plugin.buycourses
+ */
+require_once '../config.php';
+
+$plugin = BuyCoursesPlugin::create();
+$cecabankEnabled = $plugin->get('cecabank_enable') === 'true';
+
+if (!$cecabankEnabled) {
+    api_not_allowed(true);
+}
+
+$reference = $_POST['Num_operacion'];
+$receivedAmount = $_POST['Importe'];
+$receivedSignature = $_POST['Firma'];
```

Comment:
If you are sure that those variables will always be handled safely, please comment above these 3 lines something like `// Safe POST variables`.
I personally prefer it when you apply some kind of filter directly. `Num_operacion` is probably an int, `Importe` is probably a float and `Firma` is probably some kind of string (so no filter there unless you know the format of that signature and you can check it here).
The general rule of thumb is that, if the variable is unfiltered, you should use the $_POST variable name directly (in the function call below, for example), not put it in a local variable that makes you believe it is safe because it has been filtered.

Reply:
So, in short, use the $_POST['var'] name rather than assigning it to another, local, variable, then add the parameters typing to those functions to be sure you are not accepting something unexpected.
To avoid PHP firing up errors, you could convert the receivedAmount to (float), so you're sure the right type is sent to the method.

Reply:
I think that I understand that logic, but I´m no sure that I applied it corretly on my last push ... coul you check it?